lib/reversetunnel/conn.go

```diff
@@ -154,13 +157,33 @@ func (c *remoteConn) ChannelConn(channel ssh.Channel) net.Conn {
 	return sshutils.NewChConn(c.sconn, channel)
 }
 
+func (c *remoteConn) incrementActiveSessions() {
+	atomic.AddInt64(&c.sessions, 1)
+}
+
+func (c *remoteConn) decrementActiveSessions() {
+	atomic.AddInt64(&c.sessions, -1)
+}
+
+func (c *remoteConn) activeSessions() int64 {
+	return atomic.LoadInt64(&c.sessions)
+}
+
 func (c *remoteConn) markInvalid(err error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
 	c.lastError = err
 	atomic.StoreInt32(&c.invalid, 1)
-	c.log.Debugf("Disconnecting connection to %v %v: %v.", c.clusterName, c.conn.RemoteAddr(), err)
+	c.log.Debugf("Unhealthy connection to %v %v: %v.", c.clusterName, c.conn.RemoteAddr(), err)
```

Comment:
What are your thoughts on changing the log level here? I feel like this would be a useful bit of debugging and shouldn't be too chatty (and if it is, it's likely indicative of an issue).